backend/src/chats/chat/chat.controller.ts

```diff
@@ -39,7 +54,53 @@ export class ChatController {
 			const hashed = await bcrypt.hash(createChatDto.password, 11);
 			createChatDto.password = hashed;
 		}
-		return this.chatService.insert(createChatDto);
+
+		// post can contain minimal data:
+		// users: [{"id":7,"permissions":["read","post"]}]
+		const users: UserInChat[] = [];
+		if (createChatDto.hasOwnProperty('users')) {
+			createChatDto.users.forEach((user) => users.push(user));
+			delete createChatDto.users;
+		}
+
+		let chat: Chat = await this.chatService.save(createChatDto);
+		if (users.length) {
+			const relations: Partial<ChatUserPermission>[] = [];
+			users.forEach((user: UserInChat) => {
+				user.permissions.forEach((permission: string) => {
+					relations.push({
+						user_id: user.id,
+						chat_id: chat.id,
+						permission: permission,
+					});
+				});
+			});
+			const rels = await this.chatUserPermissionService.save(relations);
+
+			chat = await this.chatService.findOne({
+				where: { id: chat.id },
+				relations: { has_users: { users: true } },
+			});
+		}
+
+		if (this.socketService.chatServer !== null) {
+			const socketMessage: SocketMessage<Chat_List_Item> = {
+				action: 'new',
+				data: {
+					id: chat.id,
+					name: chat.name,
+					type: chat.type as Chat_Type,
+					users:
+						chat.users?.map((user) => ({
+							id: user.id,
+							name: user.name,
+							avatar: user.avatar,
+						})) ?? [],
+				},
+			};
+			this.socketService.chatlist_emit('all', socketMessage);
+		}
+		return chat;
 	}
 
 	@Get()
```

Comment:
?

Reply:
with this comment you made me look in the file. Scan over the relevant lines. Try to figure out what you mean by the singular question mark. And then still be frustrated since I can't figure out what you mean or why you commented here....
So a little more specific please ;)

Reply:
haha I think I saw something empty?